### DISCUSSION TITLE
`build.gradle`のdependenciesはバージョン指定するべきか？

### DISCUSSION THREAD

**Discussion by h-takesg** [kusumotolab/kJuiceApp#73] [General]
今後build.gradleにおいてバージョン指定は
1. 最新を動的指定（`latest.integration`）
2. メジャーバージョンだけ固定（`1.+`）
3. 完全固定（`1.3.0`）

のどれがいいか

今は削れるところは削っている．これは極力最新のバージョンを使わせるため．最新バージョンの動的指定．
（実際は`latest.integration`とかを使うほうが正しい）

一方で依存先のバージョンは固定すべきともいえる（e.g. npmの`package-lock.json`）

動的指定のProCon
Pro
+ ほっといてもビルド毎にそこそこ最新の状態を維持できる

Con
- 環境の再現性がない

プロジェクト内で言うとdevcontainerが用意するjavaやnode環境，docker-composeのイメージはメジャー番号の指定まで．

**Comment by Haur514** [comment]
2か3がいいのかなという気がしてる．
ライブラリの最新バージョンが変わったことによるAPIの変更を手動で把握することは難しいとの，楠本研究室OBの方からの研究報告があるので．
